docs/cloud/azure/architecture.md

```diff
@@ -1 +1,937 @@
+
+General TCP load balancers with an HDB pool offer better performance
+than a stand-alone HDB, however, utilizing the underlying HDBs is not
+optimal. Consider three clients C1, C2, C3 and two servers HDB1 and
+HDB2. C1 is directed to HDB1 when establishing the TCP connection, C2 to
+HDB2 and C3 to HDB1 again. If C1 and C3 send heavy queries and C2 sends
+a few lightweight queries, then HDB1 is overloaded and HDB2 is idle. To
+improve the load distribution the load balancer needs to go under the
+TCP layer and needs to understand the kdb+ protocol.
+
+## Logging
+
+Azure provides a fully managed logging service that performs at scale
+and can ingest application and system log data. Azure Monitor allows you
+to view, search and analyze system logs. It provides an easy to use and
+customizable interface so that e.g. DevOps can quickly troubleshoot
+applications.
+
+Azure Monitor Logs enables you to see all of your logs, regardless of
+their source, as a single and consistent flow of events ordered by time.
+Events are organized into log streams and each stream is part of a log
+group. Related applications typically belong to the same log group.
+
+You don’t need to modify your tick scripts to enjoy the benefits of
+Azure Monitor. The Azure Monitor agent can be installed and configured
+to forward your application log to Azure Monitor. You should use the
+policies and policy initiatives below to automatically install the agent
+and associate it to a data-collection rule, every time you create a
+virtual machine. In the host configuration file you need to provide the
+log file to watch and to which log stream the new entries should be
+sent.
+
+Almost all kdb+ tick components can benefit from cloud logging. 
+Feedhandlers log new data arrival, data and connection issues. The TP logs
+new or disappearing publishers and subscribers. It can log if the output
+queue is above a threshold. The RDB logs all steps of the EOD process
+which includes sorting and splaying of all tables. The HDB and gateway
+can log every user query.
+
+Kdb+ users often prefer to save log messages in kdb+ tables. Tables that
+are unlikely to change are specified by a schema, while entries that
+require more flexibility use key-value columns. Log tables are ingested
+by log tick plans and these Ops tables are separated from the tables
+required for the business.
+
+One benefit of storing log messages is the ability to process log
+messages in qSQL. Timeseries join functions include as-of and window
+joins. For example, gateway functions are executed hundreds of times
+during the day. The gateway query executes RDB and HDB queries, often
+via a load balancer. All these components have their own log entries.
+You can simply employ Window Join to find relevant entries and perform
+aggregation to get insight into the performance characteristics of the
+execution chain. Note nothing prevents you from logging both to
+kdb+ and to Azure Monitor.
```

Comment:
Please decide how to write "window join" and be consistent. Initial capitals or not?